### PR TITLE
Fix random printf format warnings and type mismatches

### PR DESCRIPTION
**Pull request by yamt**
## Summary
see https://github.com/apache/incubator-nuttx/pull/2222

### CODE REVIEW

File: include/graphics/nxwidgets/itextbox.hxx [hunk 1]
```diff
@@ -208,7 +208,7 @@ namespace NXWidgets
      * @return position The cursor position.
      */
 
-    virtual const int getCursorPosition(void) const = 0;
+    virtual const int32_t getCursorPosition(void) const = 0;
```
Comment (xiaoxiang781216) [inline include/graphics/nxwidgets/itextbox.hxx:211]: should we change https://github.com/apache/incubator-nuttx-apps/blob/master/include/graphics/nxwidgets/cscrollingtextbox.hxx#L329 to int instead?
moveCursorToPosition at https://github.com/apache/incubator-nuttx-apps/blob/master/include/graphics/nxwidgets/cscrollingtextbox.hxx#L320 also need to change.

Comment (yamt) [reply include/graphics/nxwidgets/itextbox.hxx:211]: i dunno. i'm not familiar with this piece of code.
is there any reason to prefer int over int32_t here?

Comment (xiaoxiang781216) [reply include/graphics/nxwidgets/itextbox.hxx:211]: two reason:

1. Other function(e.g. moveCursorToPosition) use int
2. Only CScrollingTextBox use int32_t, all other place use int

Comment (yamt) [reply include/graphics/nxwidgets/itextbox.hxx:211]: ok, i will fix

Comment (yamt) [reply include/graphics/nxwidgets/itextbox.hxx]: done

**Comment by yamt** [comment]
the following nxstyle errors are expected.
```
    examples/nrf24l01_term/nrf24l01_term.c:125:33: error: Mixed case identifier found
    examples/nrf24l01_term/nrf24l01_term.c:129:15: error: Mixed case identifier found
```

also, canutils has a lot of nxstyle errors because it doesn't seem to follow nxstyle at all.

**Comment by xiaoxiang781216** [comment]
Ok, let's skip canutils nxstyle warning now.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
LGTM, @yamt I will merge this PR now. If there are more nxstyle warnings, please create a new PR to address them.

**Pull request merged**